1158/en.md

```diff
@@ -0,0 +1,75 @@
+# LOJ-1158: Anagram Division
+---
+### Abridge
+Given a string **s** and a positive integer **d**, you have to determine how many permutations of **s** are divisible by **d**.
+Notice the constrains: $1<=|s|<=10$, $1<=d<=1001$.
+
+### Approach
+This problem can be solved by using Dynamic Programming
+Let `dp[mask][remainer]` be the number of permutaions which include digits in **s** at positions identified by indexes of bit 1 in mask (from right most) and remainder when divided by **d**.
+The transition:
+dp[mask][new_remainder] = sum of all dp[old_mask][old_remainder] with old_mask is mask erased one bit 1.
```

Comment:
it would be better if you could show some diagram or table of how the mask transition happens.